### PR TITLE
Move CRD convert functionality to v1alpha1 package

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed.>

The `v1alpha1` CRD will implement the conversion webhook. To avoid circular dependencies moving the conversion to `v1alpha1` 

**Link to tracking Issue(s):** <Issue number if applicable>

Updates: #2620 

- Resolves: #issue-number

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: apis/v1beta1/opentelemetrycollector_types.go
Comment (jaronoff97) [inline apis/v1beta1/opentelemetrycollector_types.go:25]: love to see us breaking the cycle :D

File: apis/v1beta1/opentelemetrycollector_types.go
Comment (jaronoff97) [inline apis/v1beta1/opentelemetrycollector_types.go:295]: should we wait to merge this until after [this PR](https://github.com/open-telemetry/opentelemetry-operator/pull/2664)? At the very least we should wait to release this until after that PR is merged.

**Review by jaronoff97 — COMMENTED** [review]
one question/comment. otherwise this looks good

File: apis/v1beta1/opentelemetrycollector_types.go
Comment (pavolloffay) [reply apis/v1beta1/opentelemetrycollector_types.go:295]: good catch I will rename the JSON directly in this PR. The #2664 is still a draft and I would like to move this forward ASAP to unblock other work

**Review by jaronoff97 — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]
LGTM

**Review by frzifus — APPROVED** [review]

**Pull request merged**
